assets/javascripts/job_templates.js

```diff
@@ -394,7 +394,14 @@ function submitTemplateEditor(button) {
     // Reset to the minimum viable YAML if empty
     var template = editor.doc.getValue();
     if (template === '') {
-        template = "products: {}\nscenarios: {}";
+        template = "products: {}\nscenarios: {}\n";
+        editor.doc.setValue(template);
+    }
+
+    // Ensure final linebreak, as files without it often need additional
+    // handling elsewhere
+    if (template.substring(template.length - 1) != "\n") {
```

Comment:
I think `template.substr(-1) !== '\n'` is better.

Reply:
oh nice. I tried that with `substring` and didn't see that there is also `substr` which accepts a negative offset

Reply:
done